### PR TITLE
JBPM-6968: Stunner - Palette re-design

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @romartin @hasys would you mind review this PR
Thanks.

Ps.  please don't merge yet since I'm doing a couple of tests, but it's is in a good status to start testing.
thanks.

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/src/main/java/org/kie/workbench/common/stunner/core/client/components/palette/PaletteItem.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 public interface PaletteItem {
 
+    int getPriority();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-client-api/src/main/java/org/kie/workbench/common/stunner/core/client/components/palette/PaletteItem.java]: IMO I would remove the stuff about `priority`... the model already returns lists, so you can just sort as you need using streams and that's it, I would try not to add more and more complexity for the palette builders.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/components/palette/BPMNPaletteDefinitionBuilder.java [hunk 2]
```diff
@@ -87,10 +102,58 @@
...
+            return 600;
+        }
+        if (BPMNCategories.CONTAINERS.equals(id)) {
+            return 700;
+        }
+        if (BPMNCategories.SERVICE_TASKS.equals(id)) {
+            return 800;
+        }
+
+        if ("org.kie.workbench.common.stunner.bpmn.definition.BaseCatchingIntermediateEvent".equals(id)) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/components/palette/BPMNPaletteDefinitionBuilder.java]: Why not creating these Strings in the BPMNCategories, as the others? 

Also please let's try to use not type-safe code, if you need the Id for a bean in Stunner you can do this:
`BindableAdapterUtils.getDefinitionId(BaseCatchingIntermediateEvent.class)`
This way the refactoring are more safe.

Thx!

**Comment by domhanak** [comment]
Hello @wmedvede thanks for the PR. I did try the changes in Chrome, Firefox and IE11 and didn't run into anything that would require bug filled. But I encountered a couple of things I wanted to mention/talk about.

1. Closing of the palette - I expected the palette to be closed when I click some place in diagram or when I add a Node, but it stays open when I do it and there are these 2 options to close it - Click X or click the palette category Icon again. Which seems unnecessary to me.

2. Not sure about what the correct behavior is but, let's say I select a Node and then decide I do not want it - I drop it back onto palette window. Like this:
image
the Node gets dropped into Canvas and is actually added, my expectation was that it will NOT be added and my selection would get discarded. Could you please clarify which behavior is expected here? :) 

3. @romartin you are correct with the inability to create None Task type via palette, but you can still do it using QuickMenu. However, I think that allowing creation of a task without a type is a bad practice, so I would say it is OK this way. What do you think?

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @hasys @domhanak @jomarko @manstis 

Let you know that are a few already known issues here (mostly about styling) that currently affect the DMN and CM palettes. They're just styling issues, you can anywqay work with the editors as the palettes are working. But in order to be able to have this ready for the next release branch (next Mon), we've agreed with QE to report a new ticket to address the issues (in two weeks as @wmedvede will be on PTO during next week). 

So I've reported a new ticket [JBPM-7088](https://issues.jboss.org/browse/JBPM-7088), feel free to add more issues or comments, this way once the J build is ok I'll proceed to merge this. Again sorry for the styling issues, we'll address those soon, once @wmedvede is back.

@krisv FYI

Thanks!

**Comment by romartin** [comment]
See **next tasks on the palette** in [JBPM-7088](https://issues.jboss.org/browse/JBPM-7088)

**Pull request merged**